### ISSUE TITLE
Luci URL changes (token removal) break Material theme

### ISSUE THREAD

**Issue by hnyman** [LuttyYang/luci-theme-material#18]
Luci addressing has been changed in the recent commits to remove the security tokens from the URL:
https://192.168.1.1/cgi-bin/luci/;stok=786764efae47b9552193a139cd37f5c8
-->
https://192.168.1.1/cgi-bin/luci/

There have been several commits preparing the pages for the change, but the commit https://github.com/openwrt/luci/commit/86326e0deff92a485ffd47e22ac70194abb3fd66 finally removed the tokens from the URL.

After the change Material theme is broken in Firefox.
* There is no menu on the left side on any page.
* The front page layout is broken

(EDIT: IE login is fixed) *The new system is still under development and I can login via Firefox, but not with IE11, so there are still problems in the core Luci (Issue https://github.com/openwrt/luci/issues/516 ).* So this is just a heads-up that the Luci URLs are changing, and apparently Material theme needs modifications, too.

No Menu:
![materialnomenu](https://cloud.githubusercontent.com/assets/7926856/10631022/a85d65d4-77e3-11e5-96d4-56a85dd2241a.PNG)

Scrambled layout:
![scrambledlayout](https://cloud.githubusercontent.com/assets/7926856/10631088/31fae44c-77e4-11e5-89c5-797362ba7770.PNG)

**Comment by hnyman** [comment]
The issue is apparently using the existence of ";" in the URL to sniff if the user is logged in or still at the login screen. In the new scheme there is no ";" in the URL, so Material gets it wrong.

I made a small patch in a live router to /www/luci-static/material/js/script.js . I eliminated the sniffing for ";". This works for me otherwise ok, but there is an empty left menu visible at the login screen:

```
root@OpenWrt:/usr/lib/lua/luci# diff /rom/www/luci-static/material/js/script.js
/www/luci-static/material/js/script.js
--- /rom/www/luci-static/material/js/script.js
+++ /www/luci-static/material/js/script.js
@@ -145,9 +145,6 @@
         var ret = false;
         var getUrlNode = function (href){
             var linkPos = href.indexOf(";");
-            if (linkPos == -1){
-                return "login";
-            }else{
                 linkPos = href.indexOf("/", linkPos);
                 if (linkPos == -1){
                     return "overview";
@@ -158,7 +155,6 @@
                     else
                         return link;
                 }
-            }
         };

         var currentNode = getUrlNode(window.location.pathname);

```

**Comment by hnyman** [comment]
Jow implemented a fix at the Openwrt Luci repo:
https://github.com/openwrt/luci/commit/f9e590aa4741bf337029be3f374a33543cc3f37b

**Comment by LuttyYang** [comment]
I'm busy these days, forgive me for not fixing it in time.

I will merger it latter, thanks

**Comment by LuttyYang** [comment]
I fix it in version 0.2.13,  I created a PR to fix it.
